application/src/main/java/net/corda/v5/application/interop/parameters/TypedParameter.java

```diff
@@ -2,10 +2,19 @@
 
 import org.jetbrains.annotations.NotNull;
 
+/**
+ * A [TypedParameter] is a parameter having a name and a defined [ParameterType].
+ */
 public interface TypedParameter<T> {
+    /**
+     * @return The name of the parameter, e.g. "amount".
+     */
     @NotNull
     String getName();
 
+    /**
+     * @return The type of the parameter, e.g. [ParameterTypeLabel.STRING].
```

Comment:
A method returning `ParameterType<T>` will not return e.g. `ParameterTypeLabel.STRING`, as `ParameterTypeLabel` doesn't inherit/extends from `ParameterType`